WordPress/gutenberg
# Allow directly setting a background to Cover

closes https://github.com/WordPress/gutenberg/issues/16996
closes #17131

## Description
This PR introduces color choices to cover block, essentially implementing https://github.com/WordPress/gutenberg/issues/16996
It does, however, introduces new changes to other components in order for this to happen.

- it adds `append` prop (can someone suggests a better name?) to `placeholder` & `media-placeholder` as a way to add components to the bottom of the placeholder, this could be useful to add other notices, control (like our case) or anything to extend it, it didn't make sense to filter it and rebuild the media placeholder all again for a simple thing.

- it adds `clearable` to `colorPalette` as a way to toggle the clearing button from the palette, this is useful for one time selections since there is no use for the clear button in the placeholder.

- it adds the colors to the cover block

- it plays around with showing/hiding the settings based on what is set (color or image or both)
so it disables focal point, position & overlay opacity if there is no image, the same original behavior is kept if no color is selected. 

## Introduced Gotchas
those gotchas are not problems in the implementation but things that now will cause a problem:

- now that we can have a color independent from the image, we don't have a way to remove the image after setting it, the only way is to delete the cover and creating a new one, I didn't implement this, there is only replacing the image.

- relating to the first gotcha, setting an image after setting the color is a bit implicit, but the work in https://github.com/WordPress/gutenberg/issues/11952 should help with this.
 
## How has this been tested?
- create a new cover
- set a color
- check the frontend

## Screenshots 
![ezgif com-optimize (2)](https://user-images.githubusercontent.com/6165348/63057705-c34cfe00-bee2-11e9-9d06-8922032f3708.gif)


## Types of changes





## Checklist:
